compiler/qsc_fir/src/fir.rs

```diff
@@ -310,6 +313,239 @@ impl Display for Res {
     }
 }
 
+/// A global item.
+pub enum Global<'a> {
+    /// A global callable.
+    Callable(&'a CallableDecl),
+    /// A global user-defined type.
+    Udt,
+}
+
+/// A unique identifier for an item within a package store.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct StoreItemId {
```

Comment:
Since we're going with names that parallel the corresponding HIR types (which I like), can we call this just `ItemId` to mirror `hir::ItemId`?

Reply:
There's already an `fir::ItemId` type, which is analogous to `hir::ItemId`. However, there is a slight difference between `fir::StoreItemId` and `fir::ItemId` which is the `package` field (one is an `Option` and the other one is not).

I think in this case it makes sense to keep `StoreItemId` for the additional type safety of always having a `package` field that is not an `Option`.